tool/movement.re

```diff
@@ -0,0 +1,118 @@
+enum FlyState {
+    None,
+    Up,
+    Down,
+}
+
+static mut FLYING_UP_DOWN_VELOCITY_LABEL = Text { text: "Up/Down Flying Velocity" };
+static mut FLYING_FORWARD_BACKWARD_VELOCITY_LABEL = Text { text: "Forward/Backward Flying Velocity" };
+
+static mut MOVEMENT_STATE = MovementState {
+    enable_fly: false,
+    fly_down_up_velocity: 300.,
+    fly_state: FlyState::None,
+};
+
+struct MovementState {
+    enable_fly: bool,
+    fly_down_up_velocity: float,
+    fly_state: FlyState,
+}
+
+fn create_movement_menu() -> Ui {
+    Ui::new("Movement:", List::of(
+        UiElement::Button(UiButton {
+            label: Text { text: "Enable Flying" },
+            onclick: fn(label: Text) {
+                MOVEMENT_STATE.enable_fly = true;
```

Comment:
I think it makes more sense to set the Tas movement-mode here once instead of every single tick.